cmake/std/PullRequestLinuxDriverTest.py

```diff
@@ -42,706 +40,221 @@
-    TODO: Needs explanation.
-          - How is environMap structured?
-          - How is moduleMap structured?
-    TODO: wcm: Could it be possible to offload environMap and moduleMap into a
-          JSON or Yaml file?  (Yaml allows comments which is more helpful
-          than JSON IMO.)
+    Returns:
+        Namespace object containing the arguments from the command line.
     """
+    parser    = argparse.ArgumentParser(description='Parse the repo and build information', add_help=False)
+    required  = parser.add_argument_group('Required Arguments')
+    optional  = parser.add_argument_group('Optional Arguments')
+    cwd       = os.getcwd()
+
+    default_workspace = cwd
+    if "WORKSPACE" in os.environ.keys():
+        default_workspace = os.environ["WORKSPACE"]
+
+    default_package_enables = os.path.join("..", "packageEnables.cmake")
+    default_subprojects_file = os.path.join("..", "package_subproject_list.cmake")
+    #default_package_enables = os.path.join(default_workspace, "packageEnables.cmake")
+
+
+    required.add_argument('--sourceRepo',
+                          dest="sourceRepo",
+                          action='store',
+                          help='Repo with the new changes',
+                          required=True)
+
+    required.add_argument('--sourceBranch',
+                          dest="sourceBranch",
+                          action='store',
+                          help='Branch with the new changes',
+                          required=True)
+
+    required.add_argument('--targetRepo',
+                          dest="targetRepo",
+                          action='store',
+                          help='Repo to merge into',
+                          required=True)
+
+    required.add_argument('--targetBranch',
+                          dest="targetBranch",
+                          action='store',
+                          help='Branch to merge into',
+                          required=True)
+
+    required.add_argument('--job_base_name',
+                          dest="job_base_name",
+                          action='store',
+                          help='The Jenkins job base name',
+                          required=True)
+
+    required.add_argument('--github_pr_number',
+                          dest="github_pr_number",
+                          action='store',
+                          help='The github PR number',
+                          required=True)
+
+    required.add_argument('--job_number',
+                          dest="job_number",
+                          action='store',
+                          help='The Jenkins build number',
+                          required=True)
+
+    optional.add_argument('--config',
+                          dest='configfile',
+                          action='store',
+                          default=os.path.join(cwd, "pr_config/pullrequest.ini"),
+                          help="Configuration file. Default=%(default)s",
```

Comment:
What does the configuration file do?  If I'm a user of this script, this help doesn't tell me enough.

Reply:
It's the path to the configuration .ini file that contains the environment specifications mapped out to PR jobs. 

I've updated the description.